Basic UI

vvt2001 opened the pull request:
tạo view thêm video
tạo view preview video

vvt2001 committed:
xoá storyboard

vvt2001 committed:
tạo view thêm video

vvt2001 committed:
tạo view preview video

-view sẽ phát video ngẫu nhiên trong thư viện
-nút play sẽ được hidden sau 3s không có tương tác màn hình
-các hiển thị cơ bản khác
-đang gặp vấn đề khi quay lại view thêm video thì video cũ vẫn được chạy

vvt2001 committed:
private outlets

vvt2001 committed:
tạm bỏ các hàm không liên quan

linhndSolar commented on Preview Video/PreviewVideoViewController.swift:
```diff
@@ -0,0 +1,151 @@
+    }
+    
+    private func hideButtonWithSchedule() {
+        hideButtonItem = DispatchWorkItem {
```
Đặt lại tên biến

linhndSolar commented on Preview Video/PreviewVideoViewController.swift:
```diff
@@ -0,0 +1,151 @@
+    }
+    
+    @IBAction private func handleTap(_ sender: UITapGestureRecognizer){
+        invalidateItem()
```
Đặt lại tên hàm

linhndSolar commented on Preview Video/PreviewVideoViewController.swift:
```diff
@@ -0,0 +1,151 @@
+        currentAsset?.getAVAsset(completionHandler: { asset in
+            //setup player
+            let playerItem = AVPlayerItem(asset: asset!)
+            let currentAsset = AVPlayer(playerItem: playerItem)
```
currentAsset là sao?

linhndSolar commented on Preview Video/PreviewVideoViewController.swift:
```diff
@@ -0,0 +1,151 @@
+    private func setVideoTimeLabel (){
+        // get video's duration
+        if currentAsset != nil{
+            let minutes = Int(currentAsset!.duration / 60)
```
Bỏ logic này vào TimeInterval extension

linhndSolar commented on Preview Video/PreviewVideoViewController.swift:
```diff
@@ -0,0 +1,151 @@
+    }
+    
+    private func setupPlayAndPauseButton(){
+        playAndPauseButton.layer.cornerRadius = 0.46 * playAndPauseButton.frame.width
```
0.46 là gì vậy?

linhndSolar commented on Preview Video/UISliderExtension.swift:
```diff
@@ -0,0 +1,39 @@
+        thumbView.layer.cornerRadius = radius / 2
+        
+        // Convert thumbView to UIImage
+        let renderer = UIGraphicsImageRenderer(bounds: thumbView.bounds)
```
Không làm như này, đây là render 1 view thành UIImage rồi

vvt2001 replied:
muốn chỉnh size của cái thumb ngoài cách thay image của thumb thì còn cách nào khác k ạ?

vvt2001 committed:
đổi tên biến và hàm

lấy asset random ở bên viewcontroller rồi mới truyền sang previewvideocontroller

vvt2001 committed:
thêm extension lấy nhãn thời gian cho timeinterval

vvt2001 committed:
thay giá trị corner radius để có được nút tròn
